src/modules/dataFetching/types.ts

```diff
@@ -52,3 +53,12 @@ export type FilterType<I> =
   | EnumFilter<I>
   | PickListFilter<I>
   | DateFilter<I>;
+
+export type GenericTableWithDataColumnDef<RowType, QueryVariables> = {
```

Comment:
Nice!
Maybe this name could be shortened a bit `DataColumnDef` or something?
And comment `"extend ColumnDef type to include additional attributes related to data fetching"`